molgenis-pathways/src/main/java/org/molgenis/pathways/WikiPathwaysController.java

```diff
@@ -0,0 +1,375 @@
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import com.google.common.collect.Multimap;
+
+@Controller
+@RequestMapping(URI)
+public class WikiPathwaysController extends MolgenisPluginController
+{
+	private static final Logger LOG = LoggerFactory.getLogger(WikiPathwaysController.class);
+
+	private static final String ID = "wikipathways";
+	public static final String URI = MolgenisPluginController.PLUGIN_URI_PREFIX + ID;
+	private static final Pattern EFFECT_PATTERN = Pattern
+			.compile("([A-Z]*\\|)(\\|*[0-9]+\\||\\|+)+([0-9A-Z]+)(\\|*)(.*)");
+
+	private static final String HOMO_SAPIENS = "Homo sapiens";
+	private final WikiPathwaysService wikiPathwaysService;
+	@Autowired
+	private DataService dataService;
+	private static final Pattern GENE_SYMBOL_PATTERN = Pattern.compile("^[0-9A-Za-z\\-]*");
+	private static final DocumentBuilderFactory DB_FACTORY = DocumentBuilderFactory.newInstance();
+
+	@Autowired
+	public WikiPathwaysController(WikiPathwaysService wikiPathwaysService)
+	{
+		super(URI);
+		this.wikiPathwaysService = wikiPathwaysService;
+	}
+
+	/**
+	 * Shows the start screen.
+	 * 
+	 * @param model
+	 *            the {@link Model} to fill
+	 * @return the view name
+	 */
+	@RequestMapping(method = GET)
+	public String init(Model model)
+	{
+		model.addAttribute("entitiesMeta", getVCFEntities());
+		return "view-pathways";
+	}
+
+	/**
+	 * Retrieves the list of VCF entities. They are recognized by the fact that they have an effect attribute.
+	 * 
+	 * @return {@link List} of {@link EntityMetaData} for the VCF entities
+	 */
+	private List<EntityMetaData> getVCFEntities()
+	{
+		return stream(dataService.getEntityNames().spliterator(), false).map(dataService::getEntityMetaData)
+				.filter(this::hasEffectAttribute).collect(toList());
+	}
+
+	/**
+	 * Determines if an entity has an effect attribute.
+	 * 
+	 * @param emd
+	 *            {@link EntityMetaData} of the entity
+	 * @return boolean indicating if the entity has an effect column
+	 */
+	private boolean hasEffectAttribute(EntityMetaData emd)
+	{
+		return emd.getAttribute("EFF") != null;
+	}
+
+	/**
+	 * Retrieves all pathways.
+	 * 
+	 * @return {@link Collection} of all {@link Pathway}s.
+	 * @throws ExecutionException
+	 *             if load from cache fails
+	 */
+	@RequestMapping(value = "/allPathways", method = POST)
+	@ResponseBody
+	public Collection<Pathway> getAllPathways() throws ExecutionException
+	{
+		return wikiPathwaysService.getAllPathways(HOMO_SAPIENS);
+	}
+
+	/**
+	 * Searches pathways.
+	 * 
+	 * @param searchTerm
+	 *            string to search for
+	 * @return {@link Collection} of all {@link Pathway}s found for searchTerm
+	 * @throws RemoteException
+	 */
+	@RequestMapping(value = "/filteredPathways", method = POST)
+	@ResponseBody
+	public Collection<Pathway> getFilteredPathways(@Valid @RequestBody String searchTerm) throws RemoteException
+	{
+		return wikiPathwaysService.getFilteredPathways(searchTerm, HOMO_SAPIENS);
```

Comment:
Throws error when the gene search is empty e.g. I search for MASP2 then i remove MASP2 and press enter again because i want to go back to the full list of pathways